lib/db_connection/connection.ex

```diff
@@ -593,11 +605,15 @@ defmodule DBConnection.Connection do
   defp start_timer(pid, timeout) do
     :erlang.start_timer(timeout, self(), {__MODULE__, pid, timeout})
   end
+  defp start_timer(_, :infinity, _), do: nil
```

Comment:
Lets reuse the existing timer function by calling `start_timer(:idle, timeout)` so the message carries some information!